### PR TITLE
PRfd - Merge PR and change PR status

### LINKED ISSUE #8176: PRfD - WebUI - Edit PR
As part of #8175, add the ability to edit the PR.

This includes:

Merge.
Close.
Change description.
Change base branch.

### PR DESCRIPTION
**Pull request by itaigilo**
Closes #8176.

## Change Description

Wire the "Merge PR" button to the API.
Wire the "Close PR" button to the API, and add a "Re-open PR" button.

Also, show PR details in Markdown.

-----

Screenshot 2024-09-23 at 12 27 43

### CODE REVIEW

File: webui/src/pages/repositories/repository/pulls/pulls.jsx
Comment (itaigilo) [inline webui/src/pages/repositories/repository/pulls/pulls.jsx:1]: Some extra UI stuff for the List page, 
Got into this PR, in order not to create another one in the PRs chain.

**Comment by N-o-Z** [comment]
@itaigilo Per discussion on the PRD - we do not support re-open ATM. This should be removed

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:48]: Let's open an issue for this and remove the comment as this is probably not something we're going to handle in the near future

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:62]: We should do something here - this is an inconsistent state which can lead to strange behavior

**Comment by itaigilo** [comment]
> @itaigilo Per discussion on the PRD - we do not support re-open ATM. This should be removed

Yeah I saw that - 
But it felt weird, that after closing all the buttons disappear.

I don't mind removing it, but it's small piece of code that's already implemented. 
Any good reason to *not* release it?

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:72]: Same as previous comment (also please use our TODO convention)

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:116]: This condition is not enough - Merge should be also disabled if no changes (due to modifications of the branches, or if any of them no longer exists)

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:112]: We don't support re-open

Comment (N-o-Z) [reply webui/src/pages/repositories/repository/pulls/pullDetails.jsx:112]: And also we shouldn't have a re-open button for Merged PRs

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Thanks!!
Please see comments

**Comment by N-o-Z** [comment]
> > @itaigilo Per discussion on the PRD - we do not support re-open ATM. This should be removed
> 
> Yeah I saw that - But it felt weird, that after closing all the buttons disappear.
> 
> I don't mind removing it, but it's small piece of code that's already implemented. Any good reason to _not_ release it?

Yes - it complicates the state machine of the PRs. We want to reduce the scope as much as possible

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (itaigilo) [reply webui/src/pages/repositories/repository/pulls/pullDetails.jsx:48]: [Issue](https://github.com/treeverse/lakeFS/issues/8210).

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:59]: We are already importing https://www.npmjs.com/package/exponential-backoff so lets use it

**Review by N-o-Z — APPROVED** [review]
Approved to unblock but please make sure to fix all comments before merging!

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (itaigilo) [reply webui/src/pages/repositories/repository/pulls/pullDetails.jsx:122]: True - 
Will be handled in a separate PR (or deferred).

**Comment by itaigilo** [comment]
All handled,
Except for the comments related to empty diffs, which will be handled in a separate PR.
Merging.

**Pull request merged**
